ScoutSuite/providers/gcp/facade/cloudstorage.py

```diff
@@ -1,7 +1,17 @@
+import asyncio
 from google.cloud import storage
 from ScoutSuite.providers.utils import run_concurrently
 
 class CloudStorageFacade:
     async def get_buckets(self, project_id: str):
         client = storage.Client(project=project_id)
-        return await run_concurrently(lambda: list(client.list_buckets()))
+        buckets = await run_concurrently(lambda: list(client.list_buckets()))
+        await asyncio.wait({asyncio.ensure_future(self._add_additional_info(bucket)) for bucket in buckets})
```

Comment:
I believe the use of a `get_and_set` type method would apply here. See the azure facades, they use that a lot.

Reply:
Neat!